theseus/geometry/lie_group.py

```diff
@@ -167,13 +167,22 @@ def _compose_jacobian(
     def _inverse_jacobian(group: "LieGroup") -> torch.Tensor:
         return -group.adjoint()
 
-    def _local_impl(self, variable2: Manifold) -> torch.Tensor:
+    def _local_impl(
+        self, variable2: Manifold, jacobians: List[torch.Tensor] = None
+    ) -> torch.Tensor:
         variable2 = cast(LieGroup, variable2)
-        return self.between(variable2).log_map()
+        diff = self.between(variable2)
 
-    def _local_jacobian(self, variable2: Manifold) -> Tuple[torch.Tensor, torch.Tensor]:
-        # Need the log_map derivative which is not yet implemented
-        raise NotImplementedError
+        if jacobians is not None:
+            LieGroup._check_jacobians_list(jacobians)
+            dlog: List[torch.Tensor] = []
+            ret = diff.log_map(dlog)
+            jacobians.append(-diff.inverse().adjoint() @ dlog[0])
```

Comment:
The output shape of `SO2.adjoint()` is `(N,1,1)`, whereas that of `SO2.log_map()` is `(N,1)`, which might raise errors here. @luisenp